reactioncommerce/reaction
# Permissions Fix: Use passed in "audience" param in ReactionApps

Resolves #2231 and #2234 

The issues that came as a result of work in #2145.

**How to Test:**
- Go through checkout process as a guest user and confirm that the payment section shows enabled providers (#2231)
- Also confirm that non-logged in users see search icon in the navbar (#2234)
- As admin, test granting an account access to the dashboard and confirm that dashboard items show up based on what is toggled on for that account (#1651)

.
**FYI: Why this change:**

So..... essentially, the way it was before is that, `Reaction.Apps()` expects that an `audience` field gets passed to it.... Meaning that all calls to ReactionApps will need to pass in "audience" field for proper permission checks to happen (of which not all current calls do).

In PR #2145, ReactionApps is made to "enforce" audience by reading it internally. This would have been ideally, but would need an auditing of all Reaction.App calls to meet the new standard.

Reverting that line in this PR and specifically setting audience on Reaction.App call made for the dashboard items (which is main aim around PR #2145). 

This resolves the breaking changes.... while a larger discussion around audience usage to be done later.
